### ISSUE TITLE
Kill previous task

### ISSUE THREAD

**Issue by Gioppix** [zed-industries/zed#10841]
### Check for existing issues

- [X] Completed

### Describe the feature

"allow_concurrent_runs" should provide the possibility to kill the previous task, instead of just opening a new one or waiting. This is very useful for programs that continuously run (e.g. asking for user input). 
As of now the only solution I found is manually terminating a run to allow for the new one in the queue to execute.

### If applicable, add mockups / screenshots to help present your vision of the feature

"allow_concurrent_runs": "allow_multiple" | "wait" | "terminate"

**Comment by SomeoneToIgnore** [comment]
Such behavior is available, but might not be obvious enough:
```
"allow_concurrent_runs": true,
"use_new_terminal": false
```

This combination will allow not to wait for previous tasks before spawning the new one, but forced to reuse the terminal of the previous task, which should result in the previous terminal (and the task running in it) being killed, and the new task instantly started.

**Comment by Gioppix** [comment]
@SomeoneToIgnore yeah "concurrent" seems very different from killing the previous task. Thanks tho it works great!
